core/controllers/collection_viewer.py

```diff
@@ -33,22 +32,18 @@ class CollectionPage(base.BaseHandler):
     @acl_decorators.can_play_collection
     def get(self, collection_id):
         """Handles GET requests."""
+        collection_rights = rights_manager.get_collection_rights(
```

Comment:
shifting it after retrieving collection sounds good.

I don't quite get it. It only fetches collection and rights for one collection. Also, models for collection and collection rights are different. Can you link to the use case (arunabh's) you are talking about?